lib/coinbase/smart_contract.rb

```diff
@@ -126,6 +126,72 @@ def self.create_multi_token_contract(
       new(contract)
     end
 
+    def self.read(
+      network_id:,
+      contract_address:,
+      method:,
+      abi: nil,
+      args: nil
+    )
+      args_json = args.nil? ? 'null' : JSON.generate(args)
+      abi_json = abi.nil? ? nil : JSON.generate(abi)
+
+      request = Coinbase::Client::ReadContractRequest.new({
+                                                            'method' => method,
+                                                            'args' => args_json,
+                                                            'abi' => abi_json
+                                                          })
+
+      response = Coinbase.call_api do
+        api_client = smart_contracts_api
+        api_client.read_contract(
```

Comment:
Have we considered returning a proper object as the result of this function?

Looks like the API returns a `SolidityValue`? Thoughts on having this return a `SolidityValue` as a response?

That object could then be used as an argument into contract calls or contract invocations?

That would make it so that we could more easily support types on contract invocations.

We could use our existing shorthand, or the fully fledge solidity values for example.

Reply:
copying my other comment about this - it's a good suggestion.

we haven't thought about SolidityValue across other functions. SolidityValue in particular is more of a BE representation but makes sense to think about it as a general type for other contract-related functions. we've been thinking more about  having args be a common type across different functions (particularly for node) so we have proper type-inference

Reply:
copying my other comment for reference about this - suggestion makes sense!

we haven't thought about SolidityValue across other functions. SolidityValue in particular is more of a BE representation but makes sense to think about it as a general type for other contract-related functions. we've been thinking more about  having args be a common type across different functions (particularly for node) so we have proper type-inference